scripts/build/Platform/Linux/build_linux.sh

```diff
@@ -43,7 +43,28 @@ if [[ ! -z "$RUN_CONFIGURE" ]]; then
     echo "${CONFIGURE_CMD}" > ${LAST_CONFIGURE_CMD_FILE}
 fi
 
-eval echo [ci_build] cmake --build . --target ${CMAKE_TARGET} --config ${CONFIGURATION} -j $(grep -c processor /proc/cpuinfo) -- ${CMAKE_NATIVE_BUILD_ARGS}
-eval cmake --build . --target ${CMAKE_TARGET} --config ${CONFIGURATION} -j $(grep -c processor /proc/cpuinfo) -- ${CMAKE_NATIVE_BUILD_ARGS}
+TOTAL_MEMORY=$(cat /proc/meminfo | grep MemTotal | awk '{print $2}')
+TOTAL_CORE_COUNT=$(grep -c processor /proc/cpuinfo)
+echo "Total Memory       : $TOTAL_MEMORY"
+echo "Total Cores        : $TOTAL_CORE_COUNT"
+
+# For machines that has a core to memory ratio that could cause running out of resources, 
+# we will check for en environment variable LY_MIN_MEMORY_PER_CORE to restrict the number
+# of cores that we will allow.  If set, it will use this as the minimum anticipated memory
+# that all cores will need in order to no fall below that number. 
+# This is based on the value the total memory (in Kb) divided by the LY_MIN_MEMORY_PER_CORE 
+# (in kB, e.g. 2 Gb = 2097152 kb)
+if [[ "${LY_MIN_MEMORY_PER_CORE:-0}" -gt 0 ]]; then
+    MIN_MEMORY_PER_CORE=${LY_MIN_MEMORY_PER_CORE}
+    MAX_CORE_USAGE=$(expr $TOTAL_MEMORY / $MIN_MEMORY_PER_CORE)
+    echo "Min Memory Per Job : $MIN_MEMORY_PER_CORE"
+    echo "Max Usable Cores   : $MAX_CORE_USAGE"
+    CORE_COUNT=$MAX_CORE_USAGE
+else
+    CORE_COUNT=$TOTAL_CORE_COUNT
+fi
+
+eval echo [ci_build] cmake --build . --target ${CMAKE_TARGET} --config ${CONFIGURATION} -j $CORE_COUNT -- ${CMAKE_NATIVE_BUILD_ARGS}
+eval cmake --build . --target ${CMAKE_TARGET} --config ${CONFIGURATION} -j $CORE_COUNT -- ${CMAKE_NATIVE_BUILD_ARGS}
```

Comment:
awesome!, seems that `LY_MIN_MEMORY_PER_CORE` is not defined in the `build_config.json` file for the new jobs

Reply:
LY_MIN_MEMORY_PER_CORE was meant to be unset by the build scripts but override-able locally, so it falls back to the original behavior.

Reply:
I was expecting `LY_MIN_MEMORY_PER_CORE` to be set for the gcc jobs since those are the ones that have that restriction. Then the clang jobs would take the previous behavior.
I dont see this variable being set anywhere, that means that the gcc jobs will use all cores.
Maybe I am missing who is setting this variable now, who would that be?

Reply:
On my thread ripper, i get the memory issue as well with clang, so its not a gcc issue. Its an issue with a system that has 64 GB and 64 Cores, which at a certain point reaches a breaking point: a couple link job that was using like more than 4 GB of ram, but still having 62 compile jobs running with an average of 2 GB of ram per core). I have not seen this behavior on any jenkins builds so far.
I am okay not putting this in at all, it just means i cannot use these build scripts locally without doing a local modification on them

Reply:
If there are issues with GCC on the build nodes, then i'll set the value in the gcc jobs

Reply:
ah, gotcha, so this issue was happening in gcc but for local systems, not for Jenkins
I am fine with how this is right now, if it doesnt happen in jenkins, I would let it run there full-core, dropping the change request